### PR TITLE
Add no resources available metric

### PR DESCRIPTION
**Pull request by fdc-ntflx**
### Context

Explain context and other details for this pull request.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java [hunk 1]
```diff
@@ -454,6 +454,9 @@ private void onTaskExecutorAssignmentRequest(TaskExecutorAssignmentRequest reque
                 assignmentTimeout);
             sender().tell(matchedExecutor.get().getKey(), self());
         } else {
+            metrics.setCounter(
+                ResourceClusterActorMetrics.NO_RESOURCES_AVAILABLE,
+                TagList.create(ImmutableMap.of("resourceCluster", clusterID.getResourceID(), "workerId", request.getWorkerId().getId(), "jobCluster", request.getWorkerId().getJobCluster(), "jobId", request.getWorkerId().getJobId())));
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java:459]: nit: do we need job cluster and job id both in there? Will job id alone be sufficient?

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java [hunk 2]
```diff
@@ -454,6 +454,9 @@ private void onTaskExecutorAssignmentRequest(TaskExecutorAssignmentRequest reque
                 assignmentTimeout);
             sender().tell(matchedExecutor.get().getKey(), self());
         } else {
+            metrics.setCounter(
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java:457]: I don't see where this is being set? Am I missing something?

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java (hunk 1 above)
Comment (fdc-ntflx) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java:459]: good point. I think we can remove job cluster for now.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java (hunk 2 above)
Comment (fdc-ntflx) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java:457]: let me rename to `incrementCounter`. counters can't be "set", they get incremented,

**Review by Andyz26 — APPROVED** [review]

**Pull request merged**